ctapipe/image/extractor.py

```diff
@@ -350,9 +350,16 @@ def integration_correction(
     return correction
 
 
-def time_parameters(
-    waveforms,
-):
+@guvectorize(
+    [
+        (float64[:], float32[:], float32[:], float32[:]),
```

Comment:
In guvectorize, you need to be carefull about the order of the signatures.

The first possible one is taken, where possible also includes "can be cast to".  So you need to have the float32 before float64, otherwise it will happily cast the float32 waveform to float64:

https://numba.readthedocs.io/en/stable/user/vectorize.html#the-guvectorize-decorator

> If you pass several signatures, beware that you have to pass most specific signatures before least specific ones (e.g., single-precision floats before double-precision floats), otherwise type-based dispatching will not work as expected:

Reply:
> Also, you change a waveform into single numbers, so I think the signatures should be:
> 
> ```
> (float32[:], float32, float32, float32)
> ```
> 
> ?

Following the scheme in extract_around_peak, it should be an array. At the end, I write fwhm_arry[0]=...